src/plugins/auto/tests/unit/stateful_model_test.cpp

```diff
@@ -0,0 +1,414 @@
+        devicesSupportedLayers[deviceName] = res;
+    }
+
+    for (auto& item : devicesSupportedLayers) {
+        ON_CALL(*core,
+                query_model(::testing::Matcher<const std::shared_ptr<const ov::Model>&>(_),
+                            ::testing::Matcher<const std::string&>(StrEq(item.first)),
+                            _))
+            .WillByDefault(Return(item.second));
+    }
+
+    ON_CALL(*core,
+            compile_model(::testing::Matcher<const std::shared_ptr<const ov::Model>&>(_),
+                          ::testing::Matcher<const std::string&>(StrEq(ov::test::utils::DEVICE_CPU)),
+                          (_)))
+        .WillByDefault(Return(mockExeNetwork));
+
+    ON_CALL(*core,
+            compile_model(::testing::Matcher<const std::shared_ptr<const ov::Model>&>(_),
+                          ::testing::Matcher<const std::string&>(StrEq(ov::test::utils::DEVICE_GPU)),
+                          (_)))
+        .WillByDefault(Return(mockExeNetworkActual));
+    if (isCumulative)
+        plugin->set_device_name("MULTI");
+    else
+        plugin->set_device_name("AUTO");
+}
+
+TEST_P(StatefulModelSupportedTest, CanFilterOutCorrectTargetDeviceWithStatefulModel) {
+    metaDevices.clear();
+    int priority = 0;
+    for (auto& item : expectedCalledTimes) {
+        auto deviceName = item.first;
+        auto times = item.second;
+        DeviceInformation devInfo(deviceName, {}, -1, {}, deviceName, priority++);
+        metaDevices.push_back(devInfo);
+        if (times >= 0) {
+            EXPECT_CALL(*core,
+                        compile_model(::testing::Matcher<const std::shared_ptr<const ov::Model>&>(_),
+                                      ::testing::Matcher<const std::string&>(StrEq(deviceName)),
+                                      ::testing::Matcher<const ov::AnyMap&>(_)))
+                .Times(times);
+        }
+    }
+    int expectedTimes = expectedCalledTimes.begin()->second;
+    ov::AnyMap config = {};
+
+    if (!devicesList.empty())
+        config.insert(ov::device::priorities(devicesList));
+
+    ON_CALL(*plugin, parse_meta_devices(_, _)).WillByDefault(Return(metaDevices));
+    ON_CALL(*plugin, get_valid_device)
+        .WillByDefault([](const std::vector<DeviceInformation>& metaDevices, const std::string& netPrecision) {
+            std::list<DeviceInformation> devices(metaDevices.begin(), metaDevices.end());
+            return devices;
+        });
+
+    config.insert(ov::intel_auto::enable_runtime_fallback(false));
+    if (isCumulative) {
+        config.insert(ov::hint::performance_mode(ov::hint::PerformanceMode::CUMULATIVE_THROUGHPUT));
+    }
+    if (expectedTimes < 0) {
+        ASSERT_THROW(plugin->compile_model(model, config), ov::Exception);
+    } else {
+        ASSERT_NO_THROW(plugin->compile_model(model, config));
+    }
+}
+
+const std::vector<StatefulModelConfigParams> testConfigs = {
+    // test cases for dynamic model
+    StatefulModelConfigParams{"CPU",
```

Comment:
Please add description for each item in first test data set.

Reply:
Updated